### PR TITLE
[ticket/13674] Change MySQL native total search results calculation

### PR DESCRIPTION
**Pull request by brunoais**
This changes how the native FULLTEXT search calculates the total
match number for MySQL.
This should improve performance as there is one less query being made
and it is being searched using the technique mentioned in the manual

https://tracker.phpbb.com/browse/PHPBB3-13674

### CODE REVIEW

File: phpBB/phpbb/search/fulltext_native.php [hunk 1]
```diff
@@ -823,6 +823,13 @@ public function keyword_search($type, $fields, $terms, $sort_by_sql, $sort_key,
 			);
 		}
 
+		// if using mysql and the total result count is not calculated yet, get it from the db
+		if(!$total_results && $is_mysql)
```
Comment (bantu) [inline phpBB/phpbb/search/fulltext_native.php]: Space after if

**Comment by bantu** [comment]
@dhruvgoel92

File: phpBB/phpbb/search/fulltext_native.php (hunk 1 above)
Comment (brunoais) [inline phpBB/phpbb/search/fulltext_native.php]: Done.

**Comment by brunoais** [comment]
@bantu I started doing this because he wasn't answering. I gave him 2 weeks to answer my PM and to appear in IRC and he didn't.

**Comment by bantu** [comment]
Maybe @naderman can comment on this patch.

**Comment by brunoais** [comment]
If there's nothing until tomorrow, I'll go by an alternative way.

**Comment by dhruvgoel92** [comment]
Hi, sorry I have not been around much lately. 
https://github.com/phpbb/phpbb/blame/3.1.x/phpBB/phpbb/search/fulltext_native.php#L767
This was removed earlier, not sure why though. I will try taking a look.

**Comment by dhruvgoel92** [comment]
Found the commit when this was moved: https://github.com/phpbb/phpbb/commit/eb49218428ca3cab1776218951608bcf3aa4d944

**Comment by brunoais** [comment]
@dhruvgoel92 
What does that have to do with this?

**Comment by brunoais** [comment]
P.S. There's another reason why each part of the code is where it is. It is to allow placing events to allow manipulating the DB query in a way that is not DBMS dependent.

**Comment by brunoais** [comment]
!set WIP

**Comment by naderman** [comment]
This looks good to me I think? @dhruvgoel92 did you see anything that speaks against this?

**Comment by naderman** [comment]
@brunoais why WIP?

**Comment by brunoais** [comment]
!unset WIP

**Comment by brunoais** [comment]
I thought of wanting it to belong to something bigger, in the end....
You are right, though. This is actually complete by itself and I can build the rest on top of this.

**Comment by dhruvgoel92** [comment]
No, I looked at the code this afternoon and this change seems fine. I will merge it tomorrow afternoon :+1:

**Pull request merged**